### ISSUE TITLE
Index doesn't include all sitemap URLs from providers

### ISSUE THREAD

**Issue by joemcgill** [GoogleChromeLabs/wp-sitemaps#71]
**Describe the bug**
Currently, the sitemap index displays URLs for each sitemap provider, but not the actual sitemaps that each provider produces.

**To Reproduce**
1. Go to /sitemap.xml
2. See the following output:

```
<sitemapindex xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
	<sitemap>
		<loc>http://xml-sitemaps.local/sitemap-posts.xml</loc>
		<lastmod>2004-10-01T18:23:17+00:00</lastmod>
	</sitemap>
	<sitemap>
		<loc>http://xml-sitemaps.local/sitemap-taxonomies.xml</loc>
		<lastmod>2004-10-01T18:23:17+00:00</lastmod>
	</sitemap>
	<sitemap>
		<loc>http://xml-sitemaps.local/sitemap-users.xml</loc>
		<lastmod>2004-10-01T18:23:17+00:00</lastmod>
	</sitemap>
</sitemapindex>
```

**Expected behavior**
Instead, the sitemap index should display one sitemap entry for every public post type page, taxonomy term page, and users page. For example:

```
<sitemapindex xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
	<sitemap>
		<loc>http://xml-sitemaps.local/sitemap-posts-post-1.xml</loc>
		<lastmod>2004-10-01T18:23:17+00:00</lastmod>
	</sitemap>
	<sitemap>
		<loc>http://xml-sitemaps.local/sitemap-posts-post-2.xml</loc>
		<lastmod>2004-10-01T18:23:17+00:00</lastmod>
	</sitemap>
	<sitemap>
		<loc>http://xml-sitemaps.local/sitemap-posts-page-1.xml</loc>
		<lastmod>2004-10-01T18:23:17+00:00</lastmod>
	</sitemap>
	<sitemap>
		<loc>http://xml-sitemaps.local/sitemap-taxonomies-post_tag-1.xml</loc>
		<lastmod>2004-10-01T18:23:17+00:00</lastmod>
	</sitemap>
	<sitemap>
		<loc>http://xml-sitemaps.local/sitemap-taxonomies-category-1.xml</loc>
		<lastmod>2004-10-01T18:23:17+00:00</lastmod>
	</sitemap>
	<sitemap>
		<loc>http://xml-sitemaps.local/sitemap-users-1.xml</loc>
		<lastmod>2004-10-01T18:23:17+00:00</lastmod>
	</sitemap>
</sitemapindex>
```

**Comment by joemcgill** [comment]
Fixing lastmod behavior is already covered in #60 and can be ignored in this ticket.

For relevant discussion about this ticket, see https://github.com/GoogleChromeLabs/wp-sitemaps/pull/68#issuecomment-553606976

**Comment by joemcgill** [comment]
Blocked by #72.
